tests/unit/test_run.py

```diff
@@ -531,13 +531,33 @@ def test_set_item(scm_run):
     assert all(scm_run["model"] == ["a_iam", "b_iam", "c_iam"])
 
 
+def test_set_item_single(scm_run):
+    scm_run["model"] = "b_iam"
+    assert all(scm_run["model"] == ["b_iam", "b_iam", "b_iam"])
+
+
 def test_set_item_not_in_meta(scm_run):
     with pytest.raises(ValueError):
         scm_run["junk"] = ["hi", "bye"]
 
     scm_run["junk"] = ["hi", "bye", "ciao"]
     assert all(scm_run["junk"] == ["hi", "bye", "ciao"])
 
+    scm_run["junk"] = ["hi", "bye", "bye"]
+    assert all(scm_run["junk"] == ["hi", "bye", "bye"])
+
+
+@pytest.mark.parametrize("key", ("model", "junk"))
+def test_set_item_nan(scm_run, key):
+    scm_run[key] = ["hi", np.NaN, "bye"]
```

Comment:
No, since NaN is numeric, it is not coerced, seem [the new test I added](https://github.com/openscm/scmdata/pull/247/commits/fa82c973fa47baef0a7aecdc947680fb58ef3ab2)